docs/src/data/amplifyMock/components.tsx

```diff
@@ -0,0 +1,175 @@
+      fill="#1877F2"
+    />
+  );
+};
+
+const AmazonIcon = (): JSX.Element => {
+  return (
+    <svg
+      aria-label="Amazon icon"
+      className="amplify-icon federated-sign-in-icon"
+      viewBox="0 0 243 264"
+      xmlns="http://www.w3.org/2000/svg"
+      preserveAspectRatio="xMidYMid"
+    >
+      <path
+        d="M230.826 208.039C227.468 203.683 208.551 205.982 200.056 206.998C197.471 207.321 197.076 205.042 199.407 203.405C214.475 192.665 239.201 195.766 242.082 199.364C244.966 202.982 241.337 228.071 227.173 240.049C225.001 241.888 222.93 240.904 223.898 238.468C227.077 230.431 234.205 212.419 230.826 208.039ZM123.769 264C71.0234 264 39.0764 241.955 14.7853 217.542C9.97339 212.706 3.71799 206.296 0.311513 200.691C-1.09773 198.372 2.59096 195.022 5.04421 196.844C35.239 219.268 79.1012 239.538 122.53 239.538C151.82 239.538 188.046 227.47 217.669 214.868C222.147 212.966 222.147 219.18 221.512 221.061C221.183 222.032 206.515 236.221 186.247 247.047C167.304 257.166 143.397 264 123.769 264Z"
+        fill="#F2541B"
+      />
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M142.943 111.185C142.943 124.756 143.268 136.054 136.406 148.123C130.856 157.913 122.027 163.95 112.222 163.95C98.8288 163.95 90.9806 153.772 90.9806 138.693C90.9806 109.036 117.677 103.647 142.943 103.647V111.185ZM178.166 196.081C175.858 198.15 171.635 198.22 169.914 196.894C157.974 187.684 149.89 173.688 149.89 173.688C130.706 193.156 117.127 199 92.2879 199C62.8772 199 40 180.905 40 144.729C40 116.461 55.3552 97.2408 77.2563 87.823C96.2094 79.5256 122.684 78.0173 142.943 75.7517C142.943 75.7517 144.633 53.933 138.699 45.9806C134.098 39.8163 126.272 36.9329 119.089 36.9329C106.127 36.8829 93.61 43.9051 91.1262 57.4188C90.4136 61.2829 87.5533 64.5261 84.54 64.206L51.0823 60.5922C48.5156 60.2951 45.0381 57.6639 45.8636 53.3081C53.644 12.3684 90.7373 0 123.989 0C140.983 0 163.21 4.51651 176.608 17.3349C193.597 33.1648 191.969 54.2755 191.969 77.2722V131.51C191.969 147.835 198.768 154.987 205.151 163.775C207.376 166.953 207.886 170.714 205.04 173.032C197.902 178.999 178.166 196.081 178.166 196.081Z"
+        fill="#F2541B"
+      />
+    </svg>
+  );
+};
+
+/**
+ * mockComponents is an object containing the override header function calls for
+ * the authenticator.
+ */
+export const mockComponents = {
+  SignIn: {
+    Header() {
+      return (
+        <Flex direction="column" paddingLeft="2rem" paddingRight="2rem">
+          <Flex direction="column" paddingTop="2rem" paddingBottom="1rem">
+            <Button
+              className="federated-sign-in-button"
+              fontWeight="normal"
+              gap="1rem"
+            >
+              <AmazonIcon />
```

Comment:
I'm not very familiar with those SVGs, but I'm curious what's the difference between using them as components and using them directly as SVG images?

No matter components or images, does it make sense if we keep them in a separate file/folder so that we can re-use them in other parts of the codebase?

Reply:
Chenwei actually pointed this out last week as well. These icons are derived from FederatedSignInButtons from the Authenticator. Chenwei and I had some ideas to add these icons to the public/svg folder to reduce duplicate code and reuse them at other places as well. However, since that is out of scope, I have just copied the code from FederatedSignInButtons.